pennylane/transforms/hamiltonian_expand.py

```diff
@@ -106,26 +106,34 @@ def hamiltonian_expand(tape, group=True):
             "Passed tape must end in `qml.expval(H)`, where H is of type `qml.Hamiltonian`"
         )
 
-    if group:
-        hamiltonian.simplify()
-        return qml.transforms.measurement_grouping(tape, hamiltonian.ops, hamiltonian.coeffs)
+    if group or hamiltonian.grouping_indices is not None:
```

Comment:
There is a little pitfall here - even if the transform is called with `group=False` (as by the qnode) we still want to use grouping information if it is available, we just don't want to compute it!